Add an option to log meta data at the top level

mooski opened the pull request:
Closes #34

mgibeau commented on lib/winston-redis.js:
```diff
@@ -39,12 +39,13 @@ class Redis extends Transport {
+    this.channel           = options.pchannel || options.channel;
+    this.pattern           = options.pattern || !!options.pchannel;
+    this.metadata          = options.meta || {};
+    this.logMetaAtTopLevel = options.logMetaAtTopLevel || false;
```
What would you think of `flatMeta` instead?

mark-storefeeder commented:
I've changed the property name as you suggested and added a description of the property to the readme.

One thing I found peculiar is that the tests are behaving differently to my own test implementation. Given that I've set `flatMeta` to true, the tests don't log any meta data, whereas my own implementation logs this data as expected. I'm not sure why.

There's an obvious alternative implementation that would solve this problem (`JSON.stringify({ level, message, ...meta })`), but it seems like extra unnecessary work just for the benefit of the tests.

mooski commented:
I've changed the property name as you suggested and added a description of the property to the readme.

One thing I found peculiar is that the tests are behaving differently to my own test implementation. Given that I've set `flatMeta` to true, the tests don't log any meta data, whereas my own implementation logs this data as expected. I'm not sure why.

There's an obvious alternative implementation that would solve this problem (`JSON.stringify({ level, message, ...meta })`), but it seems like extra unnecessary work just for the benefit of the tests.

mgibeau commented:
I'm good the the changes as is, will take a look at the tests, I've been meaning to improve them for the longest time anyways :) Thanks!

mooski commented:
Thanks for merging this!

Would it be possible to push the latest version to NPM so we can consume this change?

nfedyk commented:
Thanks for doing this. Is there are plan for a release version so this feature can be utilized?

mark-storefeeder commented:
@nfedyk It's not ideal, but we needed this feature too so I took this liberty of publishing it to npm here if that helps: https://www.npmjs.com/package/@mooski/winston-redis

mooski commented:
@nfedyk It's not ideal, but we needed this feature too so I took this liberty of publishing it to npm here if that helps: https://www.npmjs.com/package/@mooski/winston-redis

mgibeau commented:
Sorry this took a while, I just published the latest version to npm: https://www.npmjs.com/package/winston-redis/v/3.1.0
